Sources/StreamChatUI/ChatMessageList/MessageComposer/ChatMessageComposerMentionCellView_Tests.swift

```diff
@@ -0,0 +1,236 @@
+    }
+
+    func test_appearanceCustomization_usingUIConfig_withUserOnline() {
+        class RectIndicator: UIView {
+            override func didMoveToSuperview() {
+                super.didMoveToSuperview()
+                backgroundColor = .green
+                widthAnchor.constraint(equalTo: heightAnchor, multiplier: 1).isActive = true
+            }
+        }
+
+        class CustomAvatarView: ChatMessageComposerMentionAvatarView {
+            override func didMoveToSuperview() {
+                super.didMoveToSuperview()
+                backgroundColor = .green
+                widthAnchor.constraint(equalTo: heightAnchor, multiplier: 1).isActive = true
+            }
+        }
+
+        var config = UIConfig()
+        config.channelList.channelListItemSubviews.onlineIndicator = RectIndicator.self
+        config.messageComposer.mentionAvatarView = CustomAvatarView.self
+
+        let view = ChatMessageComposerMentionCellView().withoutAutoresizingMaskConstraints
+        view.widthAnchor.constraint(equalToConstant: Self.defaultCellWidth).isActive = true
+
+        view.uiConfig = config
+        view.content = ("Mr Vader", "darkforce37", TestImages.vader.url, true)
+        AssertSnapshot(view, variants: [.small])
+    }
+
+    func test_appearanceCustomization_usingUIConfig_withUserOffline() {
+        class RectIndicator: UIView {
+            override func didMoveToSuperview() {
+                super.didMoveToSuperview()
+                backgroundColor = .green
+                widthAnchor.constraint(equalTo: heightAnchor, multiplier: 1).isActive = true
+            }
+        }
+
+        class CustomAvatarView: ChatMessageComposerMentionAvatarView {
+            override func didMoveToSuperview() {
+                super.didMoveToSuperview()
+                backgroundColor = .green
+                widthAnchor.constraint(equalTo: heightAnchor, multiplier: 1).isActive = true
+            }
+        }
+
+        var config = UIConfig()
+        config.channelList.channelListItemSubviews.onlineIndicator = RectIndicator.self
+        config.messageComposer.mentionAvatarView = CustomAvatarView.self
+
+        let view = ChatMessageComposerMentionCellView().withoutAutoresizingMaskConstraints
+        view.widthAnchor.constraint(equalToConstant: Self.defaultCellWidth).isActive = true
+
+        view.uiConfig = config
+        view.content = ("Mr Vader", "darkforce37", TestImages.vader.url, false)
+        AssertSnapshot(view, variants: [.small])
+    }
+
+    func test_appearanceCustomization_usingAppearanceHook_whenUserOffline() {
+        class TestView: ChatMessageComposerMentionCellView {}
+        TestView.defaultAppearance {
+            // Modify appearance
+            $0.backgroundColor = .white
+            $0.usernameLabel.textColor = .blue
+            $0.usernameTagLabel.textColor = .cyan
+            $0.textStackView.distribution = .equalSpacing
+
+            $0.removeConstraints($0.textStackView.constraints)
```

Comment:
Why? This shouldn't be needed. That's why we use default low constraint priority.

Reply:
Yeah you are right. I was overthinking this a bit and ended up with not removing these...